Extension/src/LanguageServer/settings.ts

```diff
@@ -25,139 +25,140 @@ nls.config({ messageFormat: nls.MessageFormat.bundle, bundleFormat: nls.BundleFo
 const localize: nls.LocalizeFunc = nls.loadMessageBundle();
 
 export interface Excludes {
-    [key: string]: boolean | { when: string };
+    [key: string]: (boolean | { when: string });
+}
+
+export interface Associations {
+    [key: string]: string;
 }
 
 export interface WorkspaceFolderSettingsParams {
     uri: string | undefined;
-    intelliSenseEngine: string | undefined;
-    intelliSenseEngineFallback: boolean | undefined;
-    autocomplete: string | undefined;
-    autocompleteAddParentheses: boolean | undefined;
-    errorSquiggles: string | undefined;
-    exclusionPolicy: string | undefined;
-    preferredPathSeparator: string | undefined;
-    intelliSenseCachePath: string | undefined;
-    intelliSenseCacheSize: number | undefined;
-    intelliSenseMemoryLimit: number | undefined;
-    dimInactiveRegions: boolean | undefined;
-    suggestSnippets: boolean | undefined;
-    legacyCompilerArgsBehavior: boolean | undefined;
-    defaultSystemIncludePath: string[] | undefined;
-    cppFilesExclude: Excludes | undefined;
-    clangFormatPath: string | undefined;
+    intelliSenseEngine: string;
+    intelliSenseEngineFallback: boolean;
+    autocomplete: string;
+    autocompleteAddParentheses: boolean;
+    errorSquiggles: string;
+    exclusionPolicy: string;
+    preferredPathSeparator: string;
+    intelliSenseCachePath: string;
+    intelliSenseCacheSize: number;
+    intelliSenseMemoryLimit: number;
+    dimInactiveRegions: boolean;
+    suggestSnippets: boolean;
+    legacyCompilerArgsBehavior: boolean;
+    defaultSystemIncludePath: string[];
+    cppFilesExclude: Excludes;
+    clangFormatPath: string;
     clangFormatStyle: string | undefined;
```

Comment:
I think this `| undefined` should be removed from clangFormatStyle and the usage the errors as a result should be changed to `clangFormatStyle: util.resolveVariables(settings.clangFormatStyle, this.AdditionalEnvironment),`